### ISSUE TITLE
CI: K8sDatapathConfig ManagedEtcd Check connectivity with managed etcd: Only 2/3 nodes appeared in cilium-health status

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#10565]
## K8sDatapathConfig ManagedEtcd Check connectivity with managed etcd

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/17886/testReport/Suite-k8s-1/11/K8sDatapathConfig_ManagedEtcd_Check_connectivity_with_managed_etcd/

[c7fce67f_K8sDatapathConfig_ManagedEtcd_Check_connectivity_with_managed_etcd.zip](https://github.com/cilium/cilium/files/4326233/c7fce67f_K8sDatapathConfig_ManagedEtcd_Check_connectivity_with_managed_etcd.zip)

This test seems to be failing in precheck because it is waiting for 3 nodes to be ready. Investigating the sysdump, there are only two nodes.

Maybe this is a flake caused by previous test run with three nodes?

Possibly related: #10538

### Stacktrace
```
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.11-gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:430
cilium pre-flight checks failed
Expected
    <*errors.errorString | 0xc0002d41e0>: {
        s: "CiliumPreFlightCheck error: Timeout reached: PreflightCheck failed: Last polled error: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s1 k8s2 k8s3]'",
    }
to be nil
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.11-gopath/src/github.com/cilium/cilium/test/k8sT/assertionHelpers.go:113
```
### Standard Output
```
Number of "context deadline exceeded" in logs: 0
Number of "level=error" in logs: 0
Number of "level=warning" in logs: 0
Number of "Cilium API handler panicked" in logs: 0
Number of "Goroutine took lock for more than" in logs: 0
No errors/warnings found in logs
Cilium pods: [cilium-49c7h cilium-kkqnq]
Netpols loaded: 
CiliumNetworkPolicies loaded: 
Endpoint Policy Enforcement:
Pod                              Ingress   Egress
testclient-j8t6d                 false     false
app1-798d4f944d-t8hfw            false     false
app3-68fb594d47-7pd5n            false     false
cilium-etcd-2tz7th4xb6           false     false
cilium-etcd-5tcn4n4ncp           false     false
cilium-etcd-m6g9hdq4f7           false     false
coredns-687db6485c-n8559         false     false
etcd-operator-7fcbd85cfb-w6mdx   false     false
test-k8s2-8d47f6c76-s8xxv        false     false
testclient-82fv2                 false     false
testds-7llbx                     false     false
testds-7wn7k                     false     false
app2-dc85b4585-4jbdm             false     false
Cilium agent 'cilium-49c7h': Status: Ok  Health: Ok Nodes "" ContinerRuntime:  Kubernetes: Ok KVstore: Ok Controllers: Total 60 Failed 0
Cilium agent 'cilium-kkqnq': Status: Ok  Health: Ok Nodes "" ContinerRuntime:  Kubernetes: Ok KVstore: Ok Controllers: Total 52 Failed 0
```
### Standard Error
```
STEP: Installing Cilium
STEP: Installing DNS Deployment
STEP: Performing Cilium preflight check
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-kkqnq': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s1 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-kkqnq': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s1 k8s2 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-kkqnq': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s1 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-kkqnq': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s3 k8s1 k8s2]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-kkqnq': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s3 k8s1]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-kkqnq': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s3 k8s2 k8s1]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-kkqnq': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s1 k8s3 k8s2]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-kkqnq': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s1 k8s2 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-kkqnq': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s3 k8s1]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-kkqnq': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s3 k8s1 k8s2]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-kkqnq': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s1 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-kkqnq': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s1 k8s2 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s1 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s1 k8s2 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s1 k8s3 k8s2]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s1 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s3 k8s1 k8s2]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s1 k8s3 k8s2]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s3 k8s2 k8s1]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s1 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s3 k8s1]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s3 k8s2 k8s1]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s1 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s1 k8s2 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s1 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s1 k8s3 k8s2]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s1 k8s2 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s3 k8s2 k8s1]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s1 k8s3 k8s2]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s3 k8s1 k8s2]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s3 k8s1]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s1 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s3 k8s1]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s1 k8s3 k8s2]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s1 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s3 k8s1]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s3 k8s2 k8s1]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s1 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s1 k8s2 k8s3]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s3 k8s2 k8s1]'
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s1 k8s2 k8s3]'
STEP: Performing Cilium status preflight check
=== Test Finished at 2020-03-12T18:04:06Z====
===================== TEST FAILED =====================
STEP: Performing Cilium controllers preflight check
cmd: kubectl get pods -o wide --all-namespaces
Exitcode: 0 
Stdout:
 	 NAMESPACE           NAME                                    READY   STATUS    RESTARTS   AGE   IP              NODE   NOMINATED NODE
	 default             test-k8s2-8d47f6c76-s8xxv               2/2     Running   0          5m    10.10.1.26      k8s2   <none>
	 default             testclient-82fv2                        1/1     Running   0          5m    10.10.1.148     k8s2   <none>
	 default             testclient-j8t6d                        1/1     Running   0          5m    10.10.0.138     k8s1   <none>
	 default             testds-7llbx                            2/2     Running   0          5m    10.10.0.134     k8s1   <none>
	 default             testds-7wn7k                            2/2     Running   0          5m    10.10.1.29      k8s2   <none>
	 external-ips-test   app1-798d4f944d-t8hfw                   2/2     Running   0          34m   10.10.0.203     k8s1   <none>
	 external-ips-test   app2-dc85b4585-4jbdm                    2/2     Running   0          34m   10.10.0.228     k8s1   <none>
	 external-ips-test   app3-68fb594d47-7pd5n                   2/2     Running   0          34m   10.10.1.217     k8s2   <none>
	 external-ips-test   host-client-cr5f6                       1/1     Running   0          34m   192.168.36.11   k8s1   <none>
	 external-ips-test   host-client-vtvb6                       1/1     Running   0          34m   192.168.36.12   k8s2   <none>
	 external-ips-test   host-server-1-56c9467d4b-qqbb5          2/2     Running   0          34m   192.168.36.11   k8s1   <none>
	 external-ips-test   host-server-2-b8d89c58c-x2qkz           2/2     Running   0          34m   192.168.36.11   k8s1   <none>
	 kube-system         cilium-49c7h                            1/1     Running   0          5m    192.168.36.12   k8s2   <none>
	 kube-system         cilium-etcd-2tz7th4xb6                  1/1     Running   0          4m    10.10.1.223     k8s2   <none>
	 kube-system         cilium-etcd-5tcn4n4ncp                  1/1     Running   0          4m    10.10.0.128     k8s1   <none>
	 kube-system         cilium-etcd-m6g9hdq4f7                  1/1     Running   0          3m    10.10.1.188     k8s2   <none>
	 kube-system         cilium-etcd-operator-7447867d96-jbh7h   1/1     Running   0          5m    192.168.36.12   k8s2   <none>
	 kube-system         cilium-kkqnq                            1/1     Running   0          5m    192.168.36.11   k8s1   <none>
	 kube-system         cilium-operator-8878f5b5d-8g5z7         1/1     Running   0          5m    192.168.36.12   k8s2   <none>
	 kube-system         coredns-687db6485c-n8559                1/1     Running   0          1h    10.10.0.238     k8s1   <none>
	 kube-system         etcd-k8s1                               1/1     Running   0          1h    192.168.36.11   k8s1   <none>
	 kube-system         etcd-operator-7fcbd85cfb-w6mdx          1/1     Running   0          5m    10.10.1.121     k8s2   <none>
	 kube-system         kube-apiserver-k8s1                     1/1     Running   0          1h    192.168.36.11   k8s1   <none>
	 kube-system         kube-controller-manager-k8s1            1/1     Running   0          1h    192.168.36.11   k8s1   <none>
	 kube-system         kube-scheduler-k8s1                     1/1     Running   0          1h    192.168.36.11   k8s1   <none>
	 kube-system         log-gatherer-dq4w4                      1/1     Running   0          49m   192.168.36.13   k8s3   <none>
	 kube-system         log-gatherer-qzcd7                      1/1     Running   0          49m   192.168.36.12   k8s2   <none>
	 kube-system         log-gatherer-z8v6d                      1/1     Running   0          49m   192.168.36.11   k8s1   <none>
	 kube-system         registry-adder-7nqrr                    1/1     Running   0          50m   192.168.36.13   k8s3   <none>
	 kube-system         registry-adder-xlcgv                    1/1     Running   0          50m   192.168.36.11   k8s1   <none>
	 kube-system         registry-adder-zpps6                    1/1     Running   0          50m   192.168.36.12   k8s2   <none>
	 
Stderr:
 	 

Fetching command output from pods [cilium-49c7h cilium-kkqnq]
STEP: Performing Cilium health check
Cilium is not ready yet: connectivity health is failing: cilium-agent 'cilium-49c7h': Only 2/3 nodes appeared in cilium-health status. nodes = '[k8s2 k8s3 k8s1]'
cmd: kubectl exec -n kube-system cilium-49c7h -- cilium bpf tunnel list
Exitcode: 0 
Stdout:
 	 TUNNEL              VALUE             
	 f00d::a0b:0:0:0:0   192.168.36.11:0   
	 10.10.2.0:0         192.168.36.13:0   
	 10.10.0.0:0         192.168.36.11:0   
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-49c7h -- cilium endpoint list
Exitcode: 0 
Stdout:
 	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                              IPv6                 IPv4          STATUS   
	            ENFORCEMENT        ENFORCEMENT                                                                                                              
	 24         Disabled           Disabled          41426      k8s:id=app3                                              f00d::a0c:0:0:3ddf   10.10.1.217   ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                            
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                     
	                                                            k8s:io.kubernetes.pod.namespace=external-ips-test                                                   
	 119        Disabled           Disabled          13983      k8s:io.cilium.k8s.policy.cluster=default                 f00d::a0c:0:0:8e60   10.10.1.148   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                     
	                                                            k8s:io.kubernetes.pod.namespace=default                                                             
	                                                            k8s:zgroup=testDSClient                                                                             
	 204        Disabled           Disabled          101        k8s:app=etcd                                             f00d::a0c:0:0:eb98   10.10.1.223   ready   
	                                                            k8s:etcd_cluster=cilium-etcd                                                                        
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                            
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                     
	                                                            k8s:io.cilium/app=etcd-operator                                                                     
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                         
	 769        Disabled           Disabled          13548      k8s:io.cilium.k8s.policy.cluster=default                 f00d::a0c:0:0:5645   10.10.1.26    ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                     
	                                                            k8s:io.kubernetes.pod.namespace=default                                                             
	                                                            k8s:zgroup=test-k8s2                                                                                
	 1067       Disabled           Disabled          101        k8s:app=etcd                                             f00d::a0c:0:0:f5f8   10.10.1.188   ready   
	                                                            k8s:etcd_cluster=cilium-etcd                                                                        
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                            
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                     
	                                                            k8s:io.cilium/app=etcd-operator                                                                     
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                         
	 1158       Disabled           Disabled          4          reserved:health                                          f00d::a0c:0:0:69d9   10.10.1.250   ready   
	 1821       Disabled           Disabled          100        k8s:io.cilium.k8s.policy.cluster=default                 f00d::a0c:0:0:8968   10.10.1.121   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=cilium-etcd-sa                                              
	                                                            k8s:io.cilium/app=etcd-operator                                                                     
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                         
	 2606       Disabled           Disabled          6966       k8s:io.cilium.k8s.policy.cluster=default                 f00d::a0c:0:0:2fe5   10.10.1.29    ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                     
	                                                            k8s:io.kubernetes.pod.namespace=default                                                             
	                                                            k8s:zgroup=testDS                                                                                   
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-kkqnq -- cilium bpf tunnel list
Exitcode: 0 
Stdout:
 	 TUNNEL              VALUE             
	 10.10.1.0:0         192.168.36.12:0   
	 f00d::a0c:0:0:0:0   192.168.36.12:0   
	 10.10.2.0:0         192.168.36.13:0   
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-kkqnq -- cilium endpoint list
Exitcode: 0 
Stdout:
 	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                         IPv6                 IPv4          STATUS   
	            ENFORCEMENT        ENFORCEMENT                                                                                                         
	 1051       Disabled           Disabled          6966       k8s:io.cilium.k8s.policy.cluster=default            f00d::a0b:0:0:61af   10.10.0.134   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                
	                                                            k8s:io.kubernetes.pod.namespace=default                                                        
	                                                            k8s:zgroup=testDS                                                                              
	 1133       Disabled           Disabled          10788      k8s:id=app2                                         f00d::a0b:0:0:2aa0   10.10.0.228   ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                       
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                
	                                                            k8s:io.kubernetes.pod.namespace=external-ips-test                                              
	 2093       Disabled           Disabled          101        k8s:app=etcd                                        f00d::a0b:0:0:c63d   10.10.0.128   ready   
	                                                            k8s:etcd_cluster=cilium-etcd                                                                   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                       
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                
	                                                            k8s:io.cilium/app=etcd-operator                                                                
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                    
	 2100       Disabled           Disabled          12720      k8s:id=app1                                         f00d::a0b:0:0:977c   10.10.0.203   ready   
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                       
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                
	                                                            k8s:io.kubernetes.pod.namespace=external-ips-test                                              
	 2127       Disabled           Disabled          28558      k8s:io.cilium.k8s.policy.cluster=default            f00d::a0b:0:0:54eb   10.10.0.138   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                
	                                                            k8s:io.kubernetes.pod.namespace=default                                                        
	                                                            k8s:zgroup=testDSClient                                                                        
	 3297       Disabled           Disabled          4          reserved:health                                     f00d::a0b:0:0:f828   10.10.0.141   ready   
	 3333       Disabled           Disabled          104        k8s:io.cilium.k8s.policy.cluster=default            f00d::a0b:0:0:dee9   10.10.0.238   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=coredns                                                
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                    
	                                                            k8s:k8s-app=kube-dns                                                                           
	 
Stderr:
 	 

===================== Exiting AfterFailed =====================

[[ATTACHMENT|c7fce67f_K8sDatapathConfig_ManagedEtcd_Check_connectivity_with_managed_etcd.zip]]
STEP: Installing Cilium
STEP: Installing DNS Deployment
STEP: Performing Cilium preflight check
STEP: Performing Cilium status preflight check
STEP: Performing Cilium controllers preflight check
STEP: Performing Cilium health check
STEP: Performing Cilium service preflight check
STEP: Performing K8s service preflight check
STEP: Waiting for cilium-operator to be ready
STEP: Waiting for kube-dns to be ready
STEP: Running kube-dns preflight check
STEP: Performing K8s service preflight check
```

**Comment by brb** [comment]
Just copying what I previously wrote to @aanm regarding flake:

> The test should set `opts["operator.synchronizeK8sNodes"] = "false"` to prevent the operator from propagating the CiliumNode object for k8s3.
> Could it be that the old instance of cilium-operator was running while the new instance of cilium-agent was deployed? If yes, the old cilium-operator didn't had the option set and sent the CiliumNode update?

**Comment by brb** [comment]
@aanm Could it be that https://github.com/cilium/cilium/pull/10520 fixed the flake?

**Comment by pchaigno** [comment]
> @aanm Could it be that #10520 fixed the flake?

No, we hit it again at #10019.

**Comment by tgraf** [comment]
We are hitting this issue in the master job and 4/5 builds are failing

**Comment by aanm** [comment]
fixed by #10593
